app/data_model/answer_store.py

```diff
@@ -247,6 +254,26 @@ def get_hash(self):
         """
         return hash(json.dumps(self.answers, sort_keys=True))
 
+    def upgrade(self):
+        if self.version < self._LATEST_VERSION:
+            if self.version < 1:
+                # Update Date formats
+                for answer in self.answers:
+                    block_json = SchemaHelper.get_block(g.schema_json, answer['block_id'])
+                    answer_schema_list = SchemaHelper.get_answers_by_id_for_block(block_json)
+                    answer_schema = answer_schema_list[answer['answer_id']]
+
+                    if answer_schema['type'] == 'Date':
+                        answer['value'] = datetime.strptime(answer['value'], '%d/%m/%Y').strftime('%Y-%m-%d')
+                        continue
+
+                    if answer_schema['type'] == 'MonthYearDate':
+                        answer['value'] = datetime.strptime(answer['value'], '%m/%Y').strftime('%Y-%m-%d')
```

Comment:
I think we should handle month-year dates as YYYY-MM as mentioned in https://www.w3.org/TR/NOTE-datetime. Currently:
- It's not clear when storing it with the day what the day would be
- Given just the data you don't know whether its a full date for a month-year date